### DISCUSSION TITLE
Using `transpilePackages` with a package that has un-transpiled dependencies

### DISCUSSION THREAD

**Discussion by jackleslie** [vercel/next.js#46307] [Help]
# Context

Using the [documentation example](https://nextjs.org/docs/advanced-features/compiler#module-transpilation):

```js
// next.config.js

module.exports = {
  transpilePackages: ['@acme/ui', 'lodash-es'],
}
```

Where `@acme/ui` is un-transpiled and un-bundled, and looks like this:

```json
{
  "name": "@acme/ui",
  "dependencies": {
    "@acme/ui-tools": "^1.0.0"
  }
}
```

And `@acme/ui-tools` is also un-transpiled and un-bundled.

# Questions

- What is the expected behaviour here? 
- Can we assume that `@acme/ui-tools` will also get transpiled and bundled by Next.js? 
- Or do we need to also add `@acme/ui-tools` to `transpilePackages` ourselves?

**Comment by jackleslie** [comment]
From some testing, it appears that this is the case:

> Or do we need to also add `@acme/ui-tools` to `transpilePackages` ourselves?

So we need to add all un-transpiled and un-bundles sub-dependencies to `transpilePackages` manually

Could someone confirm this?

**Comment by 2wheeh** [comment]
It works without adding un-transpiled sub-dependencies (`ui-tools`) to `transpilePackages` when `ui-tools` is a workspace. (I didn't check the case that `ui-tools` is an un-transpiled package from npm.)

The way I see it, It's natural not to add sub-dependencies to `transpilePackages` considering the responsibility of each dependency.
